Engineering/Mobile/Swift 5.4 @resultBuilder.md

```diff
@@ -0,0 +1,239 @@
+        self.distribution = .fill
+        self.spacing = 16
+        self.axis = .horizontal
+    }
+}
+
+public class UIZStack: UIView {
+    convenience init(@UIViewBuilder _ builder: () -> [UIView]) {
+        self.init()
+        builder().forEach { view in
+            self.addSubview(view)
+            view.translatesAutoresizingMaskIntoConstraints()
+            view.fitToSuperView()
+        }
+    }
+}
+```
+
+With the technique in the last article mix with `@resultBuilder` we can fully write UI using UIKit just like SwiftUI
+
+<p align="center">
+ <img src="../../_assets/ios_uikit_builder_pattern_banner.png" width="350">
+</p>
+
+```Swift
+UIZStack(spacing: 16) {
+    UIVStack(spacing: 16) {
+        UIImageView(image: UIImage(named:"banner"))
+        UIView()
+            .backgroundColor(.clear)
+            .heightAnchor(height: 20)
+        UILabel()
+            .text(title)
+            .font(UIFont.systemFont(ofSize: UIFont.largeSize))
+            .textAlignment(.center)
+            .color(.black60)
+        UILabel()
+            .text(subtitle)
+            .font(UIFont.systemFont(ofSize: UIFont.normalSize))
+            .textAlignment(.center)
+            .color(.black60)
+            .numberOfLines(0)
+        
+        UIButton()
+            .mintStyle()
+            .title("Update Now")
+            .tap(action: { [weak self] in
+                navigateToAppStore()
+            })
+            .heightAnchor(height: 44)
+    }
+}
+
+```
+
+### Bonus parts
+
+**Support `if-else` and `loop`**
+
+`@resultBuilder` have static function `buildEither` helping you add `if-else` stagement to DSL as well as `builderArray` heping you add looping to DSL.
+The define just like a `buildBlock`:
+
+```Swift
+public static func buildEither(first component: [UIView]) -> [UIView] {
+    }
+public static func buildEither(second component: [UIView]) -> [UIView] {
+    }
+public static func buildArray(_ components: [[UIView]]) -> [UIView] {
+    }
+```
+
+```Swift
+UIHStack {
+    if true {
+        UIView().backgroundColor(.red)
+    } else {
+        UIView().backgroundColor(.green)
+    }
+
+    for image in images {
+        UIImageView(image: image)
+    }
+}
+```
+
+**Config container:**
+
+Adding config to the init to settings container view.
+
+```Swift
+public class UIVStack: UIStackView {
+    public convenience init(alignment: UIViewAlignment = .center, spacing: CGFloat = 16,  @UIViewBuilder _ builder: () -> [UIView]) {
+        self.init(arrangedSubviews: builder())
+        self.distribution = .fill
+        self.spacing = spacing
+        self.alignment = alignment
+        self.axis = .vertical
+    }
+}
+
+UIHStack(alignment: .trailling, spacing: 8) {
+
+}
+```
+
+Thank you for following us to the end of this brainey, hopfully you enjoy it.
+
+If you have anyquestion feel free to email to support@dwarvesv.com.
+
+If you like it please give us a ⭐️ to bring more brainey in the futute.
+
+Happy coding 🧑‍💻!
```

Comment:
Please avoid pleasantries in your notes. Brainery notes are close to fleeting notes, which are often semantically not aware of any subject or audience outside its context.